src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,452 @@
+﻿using System;
+using System.Reflection;
+using Godot;
+
+/// <summary>
+///   Manages spawning and processing ripple effect
+/// </summary>
+public partial class MembraneWaterRipple : Node
+{
+    [Export]
+    public bool EnableEffect = true;
+
+    [Export]
+    public float RippleStrength = 0.8f;
+
+    [Export]
+    public Color WaterColor = new(0.0f, 0.0f, 0.0f, 0.02f);
+
+    [Export]
+    public float WaterSize = 18.0f; // Base water size
+
+    [Export]
+    public float Phase = 0.2f;
+
+    /*Controls how quickly ripples dissipate
+    higher values = less viscous appearance*/
+
+    [Export]
+    public float Attenuation = 0.998f;
+
+    [Export]
+    public string ShaderPath = "res://shaders/MembraneWaterRipple.gdshader";
+
+    [Export]
+    public bool DynamicScaling = true;
+
+    [Export]
+    public float ScaleCheckInterval = 0.5f;
+
+    [Export]
+    public float RadiusMultiplier = 2.5f;
+
+    [Export]
+    public int MaxMeshSubdivision = 100; // Reduced for better performane can be increased for a more detailed effect
+
+    [Export]
+    public float VerticalOffset = -0.15f; // How far below the membrane the water plane should be
+
+    // Default radius if we can't determine it from the membrane
+    private const float DEFAULT_MEMBRANE_RADIUS = 5.0f;
+    private MeshInstance3D? waterPlane;
+    private ShaderMaterial? waterMaterial;
+    private Membrane? parentMembrane;
+
+    // Trail tracking variables
+    private Vector3 lastPosition;
+    private Vector3 previousPosition;
+    private float lastMovementTime;
+    private Vector2 currentDirection = Vector2.Right;
+    private float currentSpeed;
+
+    // State tracking
+    private float lastScaleCheckTime;
+    private float lastCheckedMembraneRadius;
+    private bool isSetup;
+
+    // Called when the node enters the scene tree for the first time
+    public override void _Ready()
+    {
+        // Initialize the effect
+        SetProcess(true);
+        SetPhysicsProcess(true);
+
+        // Find the parent membrane
+        parentMembrane = GetParent<Membrane>();
+        if (parentMembrane == null)
+        {
+            GD.PrintErr("MembraneWaterRipple must be a child of a Membrane");
+            QueueFree();
+            return;
+        }
```

Comment:
This should be done the other way around, i.e. the membrane initializing the component inside it and not calling up to the parent. That's basically always a sign of bad Godot scene tree usage.